src/Emulator/Visuals/Flow/Flow.tsx

```diff
@@ -0,0 +1,162 @@
+import React, { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import ReactFlow, {
+  Controls,
+  Background,
+  addEdge,
+  removeElements,
+  Elements,
+  updateEdge,
+} from 'react-flow-renderer';
+
+import { createElements, getLayoutedElements } from './utils';
+import { DeviceInterface } from 'src/common/types';
+import { Button, ButtonGroup } from '@chakra-ui/core';
+import { useEmulator } from 'src/Emulator/EmulatorProvider';
+import { TopologyActions } from 'src/Emulator/useTopology';
+import { deviceColorClasses } from 'src/Emulator/Device/deviceColors';
+
+import ClickableNode from './ClickableNode';
+
+import './overrides.css';
+
+interface Props {
+  hosts: DeviceInterface[],
+  routers: DeviceInterface[],
+  switches: DeviceInterface[],
+
+  isTesting?: boolean,
+}
+
+const DEFAULT_BG_GAP = 16;
+
+const RightAlignedControls = styled(Controls)`
+  right: 10px;
+  left: unset !important;
+`;
+
+const CustomControls = styled(ButtonGroup)`
+  z-index: 5;
+  position: relative;
+`;
+
+/**
+ * Determines the number of the newly added device
+ */
+ export const getNextNumber = (s: string) => Number(s.slice(1)) + 1;
```

Comment:
Is it worth it to re-use deleted numbers? Eg: we have routers 0, 1, 2, 3 and delete 2 then make another router - it is currently 4 but 2 is now missing. Not sure if it makes a meaningful difference.

Reply:
I think that might be a good idea but since this has been the default behavior for a bit we should discuss changing it and make it into a separate PR/Ticket.

Reply:
Let's see if users get confused and then make ticket if they do.